src/components/utils/PositionInScreen/PositionInScreen.tsx

```diff
@@ -35,33 +41,202 @@ const PositionInScreen: ReactFCC<Props> = ({
+      childList: true,
+      subtree: true,
+    });
+
+    return () => {
+      if (overlayMutationObserverRef.current) {
+        overlayMutationObserverRef.current.disconnect();
+      }
+    };
+  }, [visible, overlayRef, calculatePosition, isMounted]);
+
+  useEffect(() => {
+    const handleWindowResize = () => {
+      if (visible) {
+        setIsVisible(false);
+      }
+    };
+
+    window.addEventListener('resize', handleWindowResize);
+
+    return () => {
+      window.removeEventListener('resize', handleWindowResize);
+    };
+  }, [visible, setIsVisible]);
+
+  useEffect(() => {
+    if (!visible) {
+      setIsMounted(false);
+
+      return;
+    }
+
+    if (!overlayRef.current) return;
+
+    let styleElement: HTMLStyleElement | null = null;
+
+    if (visible) {
+      styleElement = document.createElement('style');
+      styleElement.type = 'text/css';
+      styleElement.appendChild(document.createTextNode(scrollLockStyles));
+      document.head.appendChild(styleElement);
+
+      document.body.setAttribute('data-scroll-locked', 'true');
+      document.body.style.pointerEvents = 'none';
+      overlayRef.current.style.pointerEvents = 'auto';
+
+      requestAnimationFrame(() => {
+        setIsMounted(true);
+      });
+    } else {
+      document.body.removeAttribute('data-scroll-locked');
+      document.body.style.pointerEvents = '';
+    }
+
+    return () => {
+      document.body.removeAttribute('data-scroll-locked');
+      document.body.style.pointerEvents = '';
+
+      setIsMounted(false);
+
+      if (styleElement) document.head.removeChild(styleElement);
+    };
+  }, [overlayRef, visible]);
+
+  if (!visible) return null;
+
+  return createPortal(
     <div
-      css={container(withOverflow, sx)}
-      ref={(ref) => {
-        setWrapperRef(ref);
+      {...overlayProps}
+      ref={overlayRef}
+      style={{
+        position: 'fixed',
+        top: 0,
+        left: 0,
+        width: isPositioned && triggerWidth && hasWrapperWidth ? rem(triggerWidth) : '',
+        transform: `translate(${x}px, ${y}px)`,
+        opacity: isPositioned ? 1 : 0,
+        maxWidth: isPositioned ? `${maxWidth}px` : '',
+        maxHeight: isPositioned ? `${maxHeight}px` : '',
+        zIndex: '9999',
       }}
+      css={sx?.itemContainer}
     >
-      {parent}
-      <div
-        css={itemContainer(x, y, showTooltip, wrapperWidth, sx)}
-        className={'unique-tooltip-id'}
-        ref={(ref) => {
-          setItemRef(ref);
-        }}
-      >
-        {children}
-      </div>
-    </div>
+      {children}
+    </div>,
+    document.body
   );
-};
+}
+
+const scrollLockStyles = `
+body[data-scroll-locked] {
+  overflow: hidden !important;
+  overscroll-behavior: contain;
+  position: relative !important;
+  padding-left: 0px;
+  padding-top: 0px;
+  padding-right: 0px;
+  margin-left: 0;
+  margin-top: 0;
+  margin-right: 0px !important;
+}
+`;
 
 export default PositionInScreen;
```

Comment:
[optional] maybe split this file into more files (e.g.: Overlay, styles) to make it more readable

Reply:
lets focus on that for v5 i would say